### DISCUSSION TITLE
save PDF with the Highlights

### DISCUSSION THREAD

**Discussion by Lazaro10e0** [Cimbali/pympress#315] [Q&A]
How can I save the PDF with the "Highlight" that I have been writting through the class?
I use the button "Save" but it doesn't do anything. If I use "Save As" it saves the pdf, without the annotations I made with "Highlight mode".
Thanks!!

**Comment by Cimbali** [comment]
Hi @Lazaro10e0, currently this isn’t supported. That would mean we’d need to convert highlights to PDF annotations but it’s technically doable I think.

**Comment by MatanZ** [comment]
A very late reply, but anyway:

I forked pympress about 5 years ago, focusing on my two needs: more annotation options (freehand as in pympress, but with user selectable line width and transparency, lines, arrows, rectangles, ellipses, text, images, latex formulas) and direct support for writing tablet (wacom) using evdev interface, rather than as an emulated mouse. 

One of the features of my fork is export of the pdf with annotation to xournalpp format and to pdf (by way of writing xopp and converting it to pdf). I believe that the export code should be easy to port back to pympress 509eb43 .

Most of this is likely to work only on Linux.

Here's a screenshot of the interface. The PDF is empty here, so everything displayed is an annotation.

![image](https://github.com/user-attachments/assets/f96beb60-9da8-43c0-a7de-26f1c64b50cb)
